tests/unit/instrumentation/test_prometheus.py

```diff
@@ -0,0 +1,119 @@
+"""
+karapace - prometheus instrumentation tests
+
+Copyright (c) 2024 Aiven Ltd
+See LICENSE for details
+"""
+
+from _pytest.logging import LogCaptureFixture
+from karapace.instrumentation.prometheus import PrometheusInstrumentation
+from karapace.rapu import RestApp
+from prometheus_client import CollectorRegistry, Counter, Gauge, Histogram
+from unittest.mock import AsyncMock, call, MagicMock, patch
+
+import aiohttp.web
+import logging
+import pytest
+
+
+class TestPrometheusInstrumentation:
+    @pytest.fixture
+    def prometheus(self) -> PrometheusInstrumentation:
+        return PrometheusInstrumentation()
```

Comment:
just a matter of taste, I prefer having just plain functions in the tests, they avoid creating implicit dependencies between the tests (by the state of the class). Since you aren't adding any state to this, do you mind change this and transform the tests that are methods of the class into plain functions? I'm not strict with this, just to avoid the temptation in the future of placing a state in the class :) (and obviously we need to be careful also about global variables, it doesn't remove completely the problem)

Reply:
I understand the reasoning. If you see that in the integration tests, I do not use a Test Class. I have this attitude towards unit tests, as we are mostly testing objects in isolation, thus it makes sense to group these tests, share dependencies between related tests, etc. 
Every test in this class `TestPrometheusInstrumentation` is only relating to the prometheus changes and thus you see the fixture is visually scoped to the class body. It's kind of a rationale similar to [unittest.TestCase](https://docs.python.org/3/library/unittest.html#unittest.TestCase) but very lightweight and more of a group for all related tests.

Reply:
Imagine we had other objects which are used for the metrics within the `prometheus.py` module, then we'd have a lot of `test_` functions, some of which at some point would not be related to the test above or below, i've found this sometimes to be confusing for unit tests. 
E2E/Integration tests on the other hand mostly encapsulate the whole test path within a single `test_` or multiple related consecutive tests functions.

Reply:
I see, again no strong feeling about this :) your explaination make sense, the long result its what count, with both the approaches you can reach monsters or well written tests. I can mark this as resolved since isn't anything other than taste